London | May-2025 | Houssam Lahlah | Wireframe

HoussamLh opened the pull request:
## Learners, PR Template

Self checklist

- [ x] I have committed my files one by one, on purpose, and for a reason
- [ x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [ x] I have tested my changes
- [ x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [ x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

cjyuan reviewed (commented):
This PR has the same issue as your other PR. (You were supposed to use the "Needs review" PR).

---

Can you also take a few minutes to check this work against the PR essentials and run your code by ChatGPT? Doing so can help speed up the review process.


Once you've finished making changes, please leave a comment on this PR so I’ll get notified. Thanks!

cjyuan commented:
Code is error-free and well-indented. Good job.

1. When a wireframe is provided, our implementation should closely reflect its appearance and layout to ensure consistency with design expectations. You're off to a solid start, but the main purpose of implementing a wireframe is to translate structure and functionality, not final visual design. Visual design is usually handled by UI/graphic designers after the wireframe phase. To better align with the wireframe, can you carry out the following changes?
    - Center the page title, the short description beneath the page title, and the content in the footer
    - Align the height of the images in the second and third articles
    
2. In the PR description, can you fix the check boxes and add a brief description of this PR?

HoussamLh commented:
Thanks! I’ve now added flexbox CSS to pin the footer to the bottom. I also updated the PR description using markdown checkboxes and confirmed that the lang attribute is present.

cjyuan commented:
Changes to the PR description looks good. But these issue still ramain:

1. According to W3C Markup Validation Service, there is no errors in your code but there is a warning. Can you address the warning?

2. One of the acceptance criteria, `"The page footer is fixed to the bottom of the viewport"`, has not yet been satisfied. Can you make the necessary change? (Suggestion: Ask ChatGPT what that requirement means).

cjyuan commented on Wireframe/index.html:
```diff
@@ -2,31 +2,92 @@
+        <img
+          src="https://support.microsoft.com/images/en-us/ae7333a6-5681-45ea-b482-e0de7c81cabd"
+          alt="Wireframe example"
+        />
+        <h2>What is the purpose of a wireframe?</h2>
+        <p>
+          A <strong>wireframe</strong> serves as a visual guide that represents
+          the skeletal framework of a website or application. The primary
+          purpose of a <strong>wireframe</strong> is to establish the basic
+          structure and layout before visual design and content are added.
+        </p>
         <p>
-          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
-          voluptates. Quisquam, voluptates.
+          It acts as a blueprint, outlining the placement of elements like
+          headings, paragraphs, images, buttons, and navigation. Wireframes help
+          designers, developers, and stakeholders visualize the user experience
+          and identify potential issues early.
         </p>
-        <a href="">Read more</a>
-      </article>
+        <a href="https://www.experienceux.co.uk/faqs/what-is-wireframing/"
+          >Read more</a
+        >
+      </section>
```
Why use `<section>` instead of `<article>`? Can you find out which element is more appropriate?
For layout purpose, `<div>` is more appropriate.

HoussamLh replied:
thank  you for your guidence, I looked into it and you're right to raise this. Based on the semantics:
<article> is more appropriate here because the content can stand alone and is self-contained (e.g., an explanation of wireframes).
<section> is better used to group related content within a page that shares a heading, but doesn’t necessarily stand alone.
I’ll update this part of the HTML to use <article> instead of <section>. Thanks for pointing it out!

cjyuan replied:
Comments in PR use Markdown syntax. We need to format HTML code (or code in general) as "code block" or "inline code" or else they won't bet rendered properly. (That's why the `<article>` and `<section>` tag are not showing up in your comment.

cjyuan commented on Wireframe/index.html:
```diff
@@ -2,31 +2,92 @@
+        >
+      </section>
+
+      <section class="side-by-side">
```
If you just want a container element for layout purpose, `<div>` is more appropriate. `<section>` has special meaning and it usually has its own header.

cjyuan reviewed (commented):
Just spotted potential misuse of `<section>` element in your code.

cjyuan commented:
Changes look good. Well done!
